File: leetcode/198-house-robber.js
// https://leetcode.com/problems/house-robber/ (top-interview-questions)

/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {

  let hOne = 0;
  let hTwo = 0;

  let maxAmount;
  for (const num of nums) {
    maxAmount = Math.max(hOne+num, hTwo);
    hOne = hTwo;
    hTwo = maxAmount;
  }

  return hTwo;
};

File: leetcode/198-house-robber.py
# https://leetcode.com/problems/house-robber/ (top-interview-questions)

from typing import List

class Solution:
	def rob(self, nums: List[int]) -> int:
		pre_last, last = 0, 0
		for num in nums:
			max_ = max(last, pre_last+num)
			pre_last = last
			last = max_
		return last
